policybot/mgrs/syncmgr/mgr.go

```diff
@@ -1188,6 +1188,104 @@ func (ss *syncState) handleTestResults() error {
 	return nil
 }
 
+func (ss *syncState) handlePostSubmitTestResults() error {
+	for _, repo := range ss.mgr.reg.Repos() {
+		r, ok := ss.mgr.reg.SingleRecord(refresher.RecordType, repo.OrgAndRepo)
+		if !ok {
+			continue
+		}
+
+		tor := r.(*refresher.TestOutputRecord)
+		g := resultgatherer.TestResultGatherer{
+			Client:           ss.mgr.blobstore,
+			BucketName:       tor.BucketName,
+			PreSubmitPrefix:  tor.PreSubmitTestPath,
+			PostSubmitPrefix: tor.PostSubmitTestPath,
+		}
+
+		scope.Debugf("Getting test results for org %s", repo.OrgLogin)
+		var completedTests = make(map[string]bool)
+		ctLock := sync.RWMutex{}
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			ctLock.Lock()
+			defer ctLock.Unlock()
+			err := ss.mgr.store.QueryTestResultByDone(ss.ctx, repo.OrgLogin, repo.RepoName,
+				func(result *storage.TestResult) error {
+					completedTests[result.RunPath] = true
+					return nil
+				})
+			if err != nil {
+				scope.Warnf("Unable to fetch previous tests: %s", err)
+			}
+			wg.Done()
+		}()
+		prPaths := g.GetAllPostSubmitTestChan(ss.ctx).WithBuffer(100)
+		// I think a composition syntax would be better here...
+		errorChan := prPaths.WithContext(ss.ctx).OnError(func(e error) {
+			// TODO: this should probably be reported out or something...
+			scope.Warnf("error processing test: %s", e)
+		}).WithParallelism(50).Transform(func(interface{}) (testRunPaths interface{}, err error) {
+			tests, err := g.GetPostSubmitTests(ss.ctx, repo.OrgLogin, repo.RepoName)
```

Comment:
I think GetPostSubmitTests and GetAllPostSubmitTestChan are probably returning the same data, so we can skip the second call. The input to this scope then will just be the testPath for a single test.

Reply:
Can I delete GetAllPostSubmitTestChan?

Reply:
Actually, I would delete the others.  You want these tests listed in a channel for paralellization, so we can delete (or unexport) the GetPostSubmitTests function.

Reply:
I think these two functions are not returning the same thing. GetPostSubmitTest call the function getTest(), return a map of testname to run number. GetAllPostSubmitTestChan return the path.

Reply:
Can you give an example of the Path that is being returned?  I suspect it will include the run number...

Reply:
GetAllPostSubmitTestChan doesn't return run number, it's just "logs/". GetPostSubmitTest includes run number, for example:"logs/release-test/1/"